src/koliseo.c

```diff
@@ -2711,7 +2711,7 @@ void kls_temp_end(Koliseo_Temp *tmp_kls)
     tmp_kls->kls->t_kls = NULL;
     tmp_kls->kls->prev_offset = tmp_kls->prev_offset;
     tmp_kls->kls->offset = tmp_kls->offset;
-    tmp_kls = NULL;
+    tmp_kls = NULL; // statement with no effect TODO: Clear tmp_kls from caller
```

Comment:
> Is it really only responsibility of caller to ensure the freed temp pointer is not usable anymore?
```                                                                                                                                     
// current requirement:                                                                                                                 
kls_temp_end(tmp_kls);                                                                                                                      
tmp_kls = NULL;                                                                                                                             
```

> Should we do:
> `*tmp_kls = (Koliseo_Temp){0};`

Are you talking about clearing tmp_kls from within the `kls_temp_end` function ?
If yes, i think we would need following changes:
```
// syntax change
void kls_temp_end(Koliseo_Temp **address_tmp_kls);
void kls_temp_end(Koliseo_Temp **address_tmp_kls) {
Koliseo_Temp *tmp_kls = *address_tmp_kls;
....
....
*address_tmp_kls = NULL; // this will clear tmp_kls; and caller wouldn't have to do it
}
```

```
// Caller would then simply call
kls_temp_end(&tmp_kls);  
```

> This one I'mis something I'm not sure, better to leave it out of this PR.
> Is this line fine as it is?

I was not sure how you want to tackle this. I suggested best way i know.
I left the comment to indicate this is a known issue/requirement for this function which is left as it is for this PR